### PR TITLE
LOOP-1146 Capture recommended bolus with actual user-entered bolus

### PR DESCRIPTION
**Pull request by darinkrauss**
- https://tidepool.atlassian.net/browse/LOOP-1146
- Add context date to SetBolusUserInfo
- Store bolus dosing decision in LoopDataManager
- Associate bolus dosing decision with current context in WatchDataManager
- Add BolusDosingDecision
- Update StoredDosingDecision Codable conformance
- Capture bolus dosing decision in carb and bolus workflows
- Additional tests

### CODE REVIEW

**Comment by darinkrauss** [comment]
LoopWorkspace PR: https://github.com/tidepool-org/LoopWorkspace/pull/277

File: Loop/Extensions/DeviceDataManager+BolusEntryViewModelDelegate.swift [hunk 1]
```diff
@@ -20,10 +20,14 @@ extension DeviceDataManager: BolusEntryViewModelDelegate {
...
     
-    func addCarbEntry(_ carbEntry: NewCarbEntry, replacing replacingEntry: StoredCarbEntry?, completion: @escaping (Result<Void>) -> Void) {
+    func addCarbEntry(_ carbEntry: NewCarbEntry, replacing replacingEntry: StoredCarbEntry?, completion: @escaping (Result<StoredCarbEntry>) -> Void) {
         loopManager.addCarbEntry(carbEntry, replacing: replacingEntry, completion: completion)
     }
 
+    func storeBolusDosingDecision(_ bolusDosingDecision: BolusDosingDecision, withDate date: Date) {
+        loopManager.storeBolusDosingDecision(bolusDosingDecision, withDate: date)
+    }
+
```
Comment (nhamming) [inline Loop/Extensions/DeviceDataManager+BolusEntryViewModelDelegate.swift:30]: Should this also have a `completion` handler like the rest?

**Review by nhamming — APPROVED** [review]
Just a question about consistency. Otherwise LGTM.

**Review by ps2 — APPROVED** [review]
LGTM!

File: Loop/Extensions/DeviceDataManager+BolusEntryViewModelDelegate.swift (hunk 1 above)
Comment (darinkrauss) [reply Loop/Extensions/DeviceDataManager+BolusEntryViewModelDelegate.swift:30]: This routine is synchronous, so no completion handler is required.

**Pull request merged**